pkg/cliplugins/kubestellar/get_log/cmd/cmd.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/client-go/util/homedir"
+	"k8s.io/klog/v2"
+)
+
+var (
+	kubeconfig string
+	followFlag bool
+	linesFlag  string
+	whichArg   string
+	logger     klog.Logger
+)
+
+func New(streams genericclioptions.IOStreams) *cobra.Command {
+	getLogCmd := &cobra.Command{
+		Use:   "get-log [flags] which",
+		Short: "Get the log output from a central KubeStellar or kcp process",
+		Args:  cobra.ExactArgs(1),
+		Run:   getLogRun,
+	}
+
+	// Create a new pflag.FlagSet for custom flags
+	fs := pflag.NewFlagSet("get-log", pflag.ExitOnError)
+	fs.StringVarP(&kubeconfig, "kubeconfig", "c", "", "Path to the kubeconfig file")
+	fs.BoolVarP(&followFlag, "follow", "f", false, "Follow the log output")
+	fs.StringVar(&linesFlag, "lines", "+0", "Number of lines to display")
+
+	// Add custom pflag.FlagSet to the Cobra command
+	getLogCmd.Flags().AddFlagSet(fs)
+
+	ctx := context.Background()
+	logger := klog.FromContext(ctx)
+	ctx = klog.NewContext(ctx, logger)
+	fs.VisitAll(func(flg *pflag.Flag) {
+		logger.V(1).Info("Command line flag", flg.Name, flg.Value)
+	})
+
+	return getLogCmd
+}
+
+func getLogRun(getLogCmd *cobra.Command, args []string) {
+	whichArg = args[0]
+	ksNamespace := "kubestellar"
+	ksSelector := "app=kubestellar-server"
+
+	logfile := ""
+	switch whichArg {
+	case "kcp":
+		logfile = "kcp.log"
+	case "where-resolver":
+		logfile = "kubestellar-where-resolver-log.txt"
+	case "mailbox-controller":
+		logfile = "mailbox-controller-log.txt"
+	case "placement-translator":
+		logfile = "placement-translator-log.txt"
+	default:
+		fmt.Println("Argument 1 must be one of: kcp, where-resolver, mailbox-controller, placement-translator")
+		os.Exit(1)
+	}
+
+	cliOpts := genericclioptions.NewConfigFlags(false)
+	cliOpts.AddFlags(getLogCmd.PersistentFlags())
```

Comment:
(1) This needs to be done before cobra parses the command line, which is before the Command's Run/RunE method is invoked.

(2) I think we want to add these flags to the cobra "persistent" flag set of the root command, because they are needed by all the subcommands of `kubectl kubestellar`.